src/external_cli/cargo/metadata.rs

```diff
@@ -29,7 +30,13 @@ where
     S: AsRef<OsStr>,
 {
     let output = command().args(additional_args).output()?;
-    let metadata = serde_json::from_slice(&output.stdout)?;
+    let metadata = serde_json::from_slice(&output.stdout).with_context(|| {
+        if let Ok(stderr) = std::str::from_utf8(&output.stderr) {
+            format!("Failed to parse `cargo metadata` output: {stderr}")
+        } else {
+            format!("Failed to parse `cargo metadata` output: {:?}", output)
+        }
+    })?;
```

Comment:
I'm not sure how to reproduce the error this PR is fixing, but if the goal is to display `stderr` to the user, could we do so through the `Command`?

```rust
    let output = command().args(additional_args)
        // Display errors to the user directly.
        .stderr(Stdio::inherit())
        .output()?;
    let metadata = serde_json::from_slice(&output.stdout)
        .context("Failed to parse `cargo metadata` output")?;
```

Reply:
Oh, yeah. That's a much cleaner way to get the error message. You can reproduce the error by inserting any error that will cause the metadata command to fail into Cargo.toml (like a dependency with a non-existant version number) and running `bevy build web`